pkg/plugin/kms_v2_server.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft and contributors.  All rights reserved.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/kubernetes-kms/pkg/config"
+	"github.com/Azure/kubernetes-kms/pkg/metrics"
+	"github.com/Azure/kubernetes-kms/pkg/version"
+
+	"k8s.io/klog/v2"
+	kmsv2 "k8s.io/kms/apis/v2"
+)
+
+// KeyManagementServiceV2Server is a gRPC server.
+type KeyManagementServiceV2Server struct {
+	kvClient Client
+	reporter metrics.StatsReporter
+}
+
+// NewKMSv2Server creates an instance of the KMS Service Server with v2 apis.
+func NewKMSv2Server(pluginConfig *Config) (*KeyManagementServiceV2Server, error) {
+	cfg, err := config.GetAzureConfig(pluginConfig.ConfigFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	kvClient, err := newKeyVaultClient(
```

Comment:
nit: it would be nice to share a single client instead of making distinct instances across v1 and v2.